core/src/main/java/org/mule/runtime/core/el/DataWeaveExpressionLanguage.java

```diff
@@ -123,14 +139,19 @@ private void addEventBindings(Event event, BindingContext.Builder contextBuilder
         contextBuilder.addBinding(name, value);
       });
       contextBuilder.addBinding(VARIABLES,
-                                new TypedValue(unmodifiableMap(flowVars), fromType(flowVars.getClass())));
+                                new TypedValue<>(unmodifiableMap(flowVars), fromType(flowVars.getClass())));
       Message message = event.getMessage();
       Attributes attributes = message.getAttributes();
-      contextBuilder.addBinding(ATTRIBUTES, new TypedValue(attributes, fromType(attributes.getClass())));
+      contextBuilder.addBinding(ATTRIBUTES, new TypedValue<>(attributes, fromType(attributes.getClass())));
       contextBuilder.addBinding(PAYLOAD, message.getPayload());
-      contextBuilder.addBinding(DATA_TYPE, new TypedValue(message.getPayload().getDataType(), fromType(DataType.class)));
+      contextBuilder.addBinding(DATA_TYPE, new TypedValue<>(message.getPayload().getDataType(), fromType(DataType.class)));
+      contextBuilder.addBinding(MULE,
+                                new TypedValue<>(new MuleInstanceContext(muleContext), fromType(MuleInstanceContext.class)));
+      contextBuilder.addBinding(SERVER, new TypedValue<>(new ServerContext(), fromType(ServerContext.class)));
+      contextBuilder.addBinding(APP, new TypedValue<>(new AppContext(muleContext), fromType(AppContext.class)));
```

Comment:
AFAIK, we don't want to expose the whole registry to users. I think we should discuss having a different view for the AppContext.